cordova-lib/src/cordova/platform.js

```diff
@@ -71,102 +72,98 @@ function add(hooksRunner, projectRoot, targets, opts) {
 
     // The "platforms" dir is safe to delete, it's almost equivalent to
     // cordova platform rm <list of all platforms>
-    if ( !fs.existsSync(platformsDir)) {
+    if (!fs.existsSync(platformsDir)) {
         shell.mkdir('-p', platformsDir);
     }
 
     return hooksRunner.fire('before_platform_add', opts)
-    .then(function() {
-        return promiseutil.Q_chainmap(targets, function(target) {
+    .then(function () {
+        return promiseutil.Q_chainmap(targets, function (target) {
             // For each platform, download it and call its "create" script.
             var parts = target.split('@');
             var platform = parts[0];
             var version = parts[1];
 
-            return Q.when().then(function() {
+            return Q.when().then(function () {
                 if (!(platform in platforms)) {
-                    return getPlatformDetailsFromDir(target);
-                } else {
+                    // First, try handling 'platform' as a directory, if it fails, try handling it as a git repository
+                    return Q.fcall(function () {
+                        return util.getPlatformDetailsFromDir(target);
+                    }).fail(function (err) {
+                        // Maybe it's a git repo
```

Comment:
Will this cause mistyped directory names to give an error message about git cloning not working? Maybe first at least check if it is a valid URL before trying to clone it?